core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +255,72 @@ export class RteHelperModalComponent {
       }
     }
     this.ngbActiveModal.dismiss(true);
+    if (this.rteFieldValues) {
+      this.rteFieldValues.unsubscribe();
+    }
   }
 
   delete(): void {
     this.ngbActiveModal.dismiss(true);
-  }
-
-  disableSaveButtonForMathRte(): boolean {
-    // This method disables the save button when the Math SVG has not yet
-    // been generated but being processed.
-    if (!this.currentRteIsMathExpressionEditor) {
-      return false;
-    } else {
-      // We know that this is a math rich text component. Hence we can make the
-      // the type more specific.
-      const { value } = this.tmpCustomizationArgs[0] as Extract<
-        CustomizationArgsNameAndValueArray[number],
-        { name: 'math_content' }
-      >;
-      return value.mathExpressionSvgIsBeingProcessed || value.raw_latex === '';
+    if (this.rteFieldValues) {
+      this.rteFieldValues.unsubscribe();
     }
   }
 
-  disableSaveButtonForLinkRte(): boolean {
-    // This method disables the save button when the `text` field for the
-    // Link RTE looks like a URL but it does not match the `url`. Otherwise,
-    // creators can make the `url` a malicious website and make the `text`
-    // a safe website.
-    if (!this.currentRteIsLinkEditor) {
-      return false;
-    }
-    // We know that this is a link rich text component. Hence we can make the
-    // the type more specific.
-    const tmpCustomizationArgs = this.tmpCustomizationArgs as Extract<
-      CustomizationArgsNameAndValueArray[number],
-      { name: 'url' | 'text' }
-    >[];
-    let url: string = tmpCustomizationArgs[0].value;
-    let text: string = tmpCustomizationArgs[1].value;
+  subscribeToRteFieldChanges(): void {
+    this.rteFieldValues = this.customizationArgsForm.valueChanges.subscribe(
+      (value) => {
+        this.updateRteSaveButtonState(value);
+      });
+  }
 
-    // First check if the `text` looks like a URL.
-    const suffixes = ['.com', '.org', '.edu', '.gov'];
-    let textLooksLikeUrl = false;
-    for (const suffix of suffixes) {
-      if (text.endsWith(suffix)) {
-        textLooksLikeUrl = true;
+  updateRteSaveButtonState(value: number | string | boolean): void {
+    if (this.componentId === this.COMPONENT_ID_MATH) {
+      let rawLatex: string = value[0].raw_latex;
+      let mathExpressionSvgIsBeingProcessed:
```

Comment:
Use indentation and parens for continuation lines:

```
let mathExpressionSvgIsBeingProcessed: boolean = (
  ...);
```

Anything starting at the leftmost space should be a "new sentence". Otherwise it should be indented.

Reply:
done